packages/cozy-client/src/CozyClient.js

```diff
@@ -1156,7 +1156,18 @@ client.query(Q('io.cozy.bills'))`)
     if (!Array.isArray(data)) {
       await this.persistVirtualDocument(data, enforce)
     } else {
-      for (const document of data) {
+      const documentsToPersist = data.filter(document => {
+        if (!document || document.cozyLocalOnly) {
+          return false
+        }
+
+        if ((!document.meta?.rev && !document._rev) || enforce) {
+          return true
+        }
+
+        return false
```

Comment:
nit: I extracted this code from `persistVirtualDocument` method, I'm wondering if we can mutualize the code in a `shouldBePersisted()` method and use it both in `persistVirtualDocument` method and here.

Reply:
Done in [ba92fe9](https://github.com/cozy/cozy-client/pull/1596/commits/ba92fe9d274529972ebaf158528deadd434c8195) 
I also moved those function  in another file, as it seems to me it was legit to have it in a dedicated section.
I also added some tests:  [677de81](https://github.com/cozy/cozy-client/pull/1596/commits/677de8137491d0b2c628e636b96b7c0e72fde2fe)